### PR TITLE
feat: integrate `go-dag-jose` plugin

### PR DESCRIPTION
**Pull request by smrz2001**
# Integrate `go-dag-jose` plugin

## Description

This PR integrates the `go-dag-jose` plugin into IPFS.

## How Has This Been Tested?

- [x] `ipfs dag put --store-codec dag-jose` and `ipfs dag get` work correctly

## Definition of Done

- [x] My code follows conventions, is well commented, and easy to understand
- [x] My code builds and tests pass without any errors or warnings
- [x] I have tagged the relevant reviewers
- [x] The changes have been communicated to interested parties
- [ ] Use the released version of `go-dag-jose`

## References:

- https://github.com/ceramicnetwork/go-dag-jose/pull/23

### CODE REVIEW

**Review by oed — APPROVED** [review]
lgtm. Should this not be a PR against ipfs/go-ipfs?

**Comment by smrz2001** [comment]
> lgtm. Should this not be a PR against ipfs/go-ipfs?

Yes, that's the plan once we test `go-ipfs` and `go-dag-jose` end-to-end. Needed a PR/commit to connect the `go-ipfs-daemon` build to but didn't want to create noise in `go-ipfs` till we were sure we were ready.

File: plugin/plugins/dagjose/dagjose.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+// Plugins is exported list of plugins that will be loaded
+var Plugins = []plugin.Plugin{
+	&dagjosePlugin{},
+}
+
+type dagjosePlugin struct{}
+
+var _ plugin.PluginIPLD = (*dagjosePlugin)(nil)
```
Comment (glcohen) [inline plugin/plugins/dagjose/dagjose.go:18]: I'm curious how you knew to add this? I always see people recommend it but never quite got it

**Review by glcohen — APPROVED** [review]

File: plugin/plugins/dagjose/dagjose.go (hunk 1 above)
Comment (smrz2001) [reply plugin/plugins/dagjose/dagjose.go:18]: Eric added this file and I built on top of his code so this was already in there.

However, after failing to divine the purpose of such code for a while, I realized that it's a great way to have the compiler (or IDE) report failures to match an interface (particularly complex ones) right up front.

For example, in one of the gigantic generated code files in `go-dag-jose`, I can see exactly what's missing in an interface implementation through a similar statement, versus finding out during compilation at some other point in the code.

Not sure if that's the actual/only purpose but that's the one that made sense to me 😄 

Screenshot (8)

Comment (glcohen) [reply plugin/plugins/dagjose/dagjose.go:18]: ahh thanks! these never show up in my IDE

**Comment by smrz2001** [comment]
Thanks for your reviews @oed and @glcohen. Closing this PR in favor of one directly against upstream.
https://github.com/ipfs/go-ipfs/pull/8569

**Pull request closed without merging**